nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java

```diff
@@ -549,7 +550,9 @@ public void process(final InputStream in) throws IOException {
         FlowFile flowFile = originalFlowFile;
         logger.info("Routing {} to {}", flowFile, REL_ORIGINAL);
         session.getProvenanceReporter().route(originalFlowFile, REL_ORIGINAL);
-        flowFile = session.putAttribute(flowFile, ROUTE_ATTRIBUTE_KEY, REL_ORIGINAL.getName());
+        if (!context.isAutoTerminated(REL_ORIGINAL)) {
+            flowFile = session.putAttribute(flowFile, ROUTE_ATTRIBUTE_KEY, REL_ORIGINAL.getName());
+        }
```

Comment:
I recommend removing this check and only retaining the conditional for the content-based processing.

Reply:
@exceptionfactory I removed the isAutoTerminated check for REL_ORIGINAL 
and retained the conditional for the content-based processing.